common/src/stack/command/stack/commands/add/pallet/__init__.py

```diff
@@ -230,23 +267,63 @@ def umount(self, iso_name, mount_point):
 
 
 	def run(self, params, args):
-		clean, stacki_pallet_dir, updatedb, run_hooks, self.username, self.password = self.fillParams([
+		clean, stacki_pallet_dir, updatedb, run_hooks, checksum, self.username, self.password = self.fillParams([
 			('clean', False),
 			('dir', '/export/stack/pallets'),
 			('updatedb', True),
 			('run_hooks', True),
+			('checksum', None),
 			('username', None),
 			('password', None),
 		])
 
 		# need to provide either both or none
-		if self.username or self.password and not all((self.username, self.password)):
+		if (self.username or self.password) and not all((self.username, self.password)):
 			raise UsageError(self, 'must supply a password along with the username')
 
 		clean = self.str2bool(clean)
 		updatedb = self.str2bool(updatedb)
 		run_hooks = self.str2bool(run_hooks)
 
+		self.checksum = {}
+		if checksum:
+			valid_checksums = ['md5', 'sha1', 'sha256', 'sha512']
+			bad_paths = []
+			for pallets in args:
+				if os.path.isdir(pallets):
+					bad_paths.append(pallets)
+
+			if bad_paths:
+				msg = f"One or more paths was specified with a checksum that is a directory. Invalid paths are: " \
+					  f"{', '.join(bad_paths)}"
+				raise UsageError(self, msg)
```

Comment:
Don't use `\` for line continuation. For strings, if you pass the parts in a function parameter without commas, they get concated for you:
```
				raise UsageError(self,
					f"One or more paths was specified with a checksum that is a directory. "
					f"Invalid paths are: {', '.join(bad_paths)}"
				)
```